### PR TITLE
23690 Move mock data into individual tests

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
The PR moves mock data from global into individual tests. The test mock data was refactored to remove all data not relevant to the test and use defaults.

## Testing done
Unit testing

## Screenshots
N/A

## Acceptance criteria
- [ ] All tests should pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx [hunk 1]
```diff
@@ -36,69 +36,10 @@ const initialState = {
...
-    },
-    id: 'vha_442GC',
-    name: 'Fort Collins VA Clinic',
-    phone: {
-      main: '970-224-1550',
-    },
-    uniqueId: '442GC',
-  },
-};
-
```
Comment (vbahinwillit) [inline src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx:88]: Global test mocked data moved to individual tests.

File: src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx [hunk 2]
```diff
@@ -36,69 +36,10 @@ const initialState = {
...
     MockDate.set(getTimezoneTestDate());
-    appointment.attributes.startDate = moment();
-    mockAppointmentInfo({
-      va: [appointment],
-      cc: [],
-      requests: [],
-      isHomepageRefresh: true,
-    });
-
-    mockFacilitiesFetch('vha_442GC', [facility]);
```
Comment (vbahinwillit) [inline src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx:101]: Moved to individual tests.

File: src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx [hunk 3]
```diff
@@ -107,11 +48,42 @@ describe('VAOS <ConfirmedAppointmentDetailsPage>', () => {
...
+    };
+
+    mockAppointmentInfo({
+      va: [appointment],
+      isHomepageRefresh: true,
+    });
+
     mockSingleAppointmentFetch({
       appointment,
-      type: 'va',
```
Comment (vbahinwillit) [inline src/applications/vaos/tests/appointment-list/components/ConfirmedAppointmentDetailsPage/index.unit.spec.jsx:112]: Not used.

**Review by jbalboni — APPROVED** [review]
LGTM

**Pull request merged**